cmd/dcs-web/search/query-rewrite.go

```diff
@@ -2,11 +2,52 @@
 package search
 
 import (
-	"fmt"
 	"net/url"
+	"regexp"
 	"strings"
 )
 
+var (
+	start = regexp.MustCompile(`(?i)^\s*(-?(?:filetype|package|pkg|path|file)):(\S+)\s+`)
```

Comment:
Figured out testing.B and did some benchmarking, putting one of the tests into a loop.

Code    | Ops     | Time/op
--------|---------|---------
strings | 5000000 | 4659 ns/op
regexp  | 2000000 | 11562 ns/op

I'll probably have to wait until the new year to revisit this.